src/sonar.ts

```diff
@@ -21,26 +21,78 @@ export type SONAR_RESULT = "issues" | "hotspots";
 
 const MAX_PAGE_SIZE = 500;
 
-export async function retrieveSonarCloudIssues(
-  sonarCloudInputs: SonarCloudInputs,
+export async function retrieveSonarIssues(
+  sonarInputs: SonarInputs,
 ): Promise<SonarSearchIssuesResult> {
-  const url = buildSonarCloudIssuesUrl(sonarCloudInputs);
-  return retrieveSonarCloudResults(sonarCloudInputs, url, "issues");
+  const url = buildSonarIssuesUrl(sonarInputs);
+  return retrieveSonarResults(sonarInputs, url, "issues");
 }
 
-export async function retrieveSonarCloudHotspots(
-  sonarCloudInputs: SonarCloudInputs,
+export function buildSonarIssuesUrl({
+  sonarHost,
+  componentKey,
+}: SonarInputs): string {
+  const { prNumber } = getGitHubContext();
+  const path = "/api/issues/search";
+
+  const queryParams = {
+    componentKeys: componentKey,
+    resolved: "false",
+    ps: MAX_PAGE_SIZE,
+    ...(prNumber && { pullRequest: prNumber }),
+  };
+
+  return buildSonarUrl({ sonarHost, path, queryParams });
+}
+
+export async function retrieveSonarHotspots(
+  sonarInputs: SonarInputs,
 ): Promise<SonarSearchHotspotResult> {
-  const url = buildSonarCloudHotspotsUrl(sonarCloudInputs);
-  return retrieveSonarCloudResults(sonarCloudInputs, url, "hotspots");
+  const url = buildSonarHotspotsUrl(sonarInputs);
+  return retrieveSonarResults(sonarInputs, url, "hotspots");
+}
+
+export function buildSonarHotspotsUrl({
```

Comment:
This one and the buildSonarIssuesUrl are almost the same besides the value of the path is possible to create a common method